integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/SearchQueryBaseIT.java

```diff
@@ -178,6 +180,34 @@ public void target_byClass_multipleTypes() {
 		} );
 	}
 
+	@Test
+	public void target_byClass_multipleTypes_entityLoadingTimeout() {
```

Comment:
See my comment on your other commit adding mapper-level tests: you need to simulate slow loading, at least.

Also, I'm sure this test currently passes (from time to time) only because your implementation rounds the timeout value to the nearest second.